### ISSUE TITLE
Handle incompatible selinux/firewall setup

### ISSUE THREAD

**Issue by robertgzr** [kinvolk/kube-spawn#77]
Right now we see the following log output if the host machine does not allow traffic forwarding from the spawned machines:
```
+ kubeadm init --skip-preflight-checks --config /etc/kubeadm/kubeadm.yml --kubernetes-version=1.7.0
[kubeadm] WARNING: kubeadm is in beta, please do not use it for production clusters.
unable to get URL "https://storage.googleapis.com/kubernetes-release/release/stable-1.7.txt": Get https://storage.googleapis.com/kubernetes-release/release/stable-1.7.txt: dial tcp: lookup storage.googleapis.com on 10.0.2.3:53: read udp 10.22.0.3:33727->10.0.2.3:53: read: no route to host
Connection to machine kube-spawn-0 terminated.
Connected to machine kube-spawn-1. Press ^] three times within 1s to exit session.
cat: /tmp/kube-spawn/token: No such file or directory
Connection to machine kube-spawn-1 terminated.
```
`kube-spawn init` does not stop/error after it failed to init the master node and tries to join the worker nodes. We should have a clear error message telling the user what the problem is here.

See #76

**Comment by blixtra** [comment]
Firewall rules were handled by https://github.com/kinvolk/kube-spawn/issues/76.

Regarding SELiunux @teg, mentioned there is a tool for generating SELinux policies from log output. Perhaps we could use that.

**Comment by teg** [comment]
I found this helpful: https://www.centos.org/docs/5/html/Deployment_Guide-en-US/sec-sel-building-policy-module.html

**Comment by dongsupark** [comment]
Ok, this issue was actually not exactly about the SELinux enforcing mode, but about informing users of exact action plans when such a firewall issue happens. It was indeed a valid issue when it was created, but right now it's not any more, as it won't happen after we merged https://github.com/kinvolk/kube-spawn/pull/110. I think we can close this issue.

OTOH as for the SELinux tips, thanks @teg for the link. Back then I already tried the tools `audit2allow` and `semodule`. I totally agree that those are great tools. Though they didn't help me make kube-spawn work with systemd-machined under SELinux enforcing mode. That's why I just had to keep turning off enforcing mode, as mentioned in https://github.com/kinvolk/kube-spawn/issues/76#issuecomment-320288720. If anyone manages to get it to work under enforcing mode, please open a new issue to discuss about that. I'd be really interested.

**Comment by alban** [comment]
> it won't happen after we merged #110. I think we can close this issue

After https://github.com/kinvolk/kube-spawn/pull/177/files#r150612474 is addressed, we can close this issue.

**Comment by alban** [comment]
The doc about disabling SELinux was merged here:
https://github.com/kinvolk/kube-spawn/pull/177/files#diff-8ed6af7314885a2fca29a17d3c758991R9

It is also mentioned here:
https://github.com/kinvolk/kube-spawn/blob/master/doc/troubleshooting.md#selinux

I don't think we will spend time on making kube-spawn work with SELinux enabled (although we'd accept PRs doing that).

Closing.
